addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/internal/api/EvohomeApiClientV2.java

```diff
@@ -0,0 +1,252 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.evohome.internal.api;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.Base64;
+import java.util.HashMap;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.openhab.binding.evohome.internal.api.models.v2.request.HeatSetPoint;
+import org.openhab.binding.evohome.internal.api.models.v2.request.Mode;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Authentication;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Location;
+import org.openhab.binding.evohome.internal.api.models.v2.response.LocationStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Locations;
+import org.openhab.binding.evohome.internal.api.models.v2.response.LocationsStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.UserAccount;
+import org.openhab.binding.evohome.internal.configuration.EvohomeAccountConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of the evohome client V2 api
+ *
+ * @author Jasper van Zuijlen - Initial contribution
+ *
+ */
+public class EvohomeApiClientV2 implements EvohomeApiClient {
+    private final Logger logger = LoggerFactory.getLogger(EvohomeApiClientV2.class);
+
+    private final SslContextFactory sslContextFactory = new SslContextFactory();
+    private final HttpClient httpClient = new HttpClient(sslContextFactory);
+
+    private EvohomeAccountConfiguration configuration = null;
+    private ApiAccess apiAccess = null;
+    private UserAccount useraccount = null;
+    private Locations locations = null;
+    private LocationsStatus locationsStatus = null;
+
+    /**
+     * Creates a new API client based on the V2 API interface
+     *
+     * @param configuration The configuration of the account to use
+     */
+    public EvohomeApiClientV2(EvohomeAccountConfiguration configuration) {
+        this.configuration = configuration;
+
+        try {
+            httpClient.start();
+        } catch (Exception e) {
+            logger.error("Could not start http client.", e);
+        }
+
+        apiAccess = new ApiAccess(httpClient);
+        apiAccess.setApplicationId(configuration.applicationId);
+    }
+
+    /**
+     * Closes the current connection to the API
+     */
+    public void close() {
+        apiAccess.setAuthentication(null);
+        useraccount = null;
+        locations = null;
+        locationsStatus = null;
+
+        if (httpClient.isStarted()) {
+            try {
+                httpClient.stop();
+            } catch (Exception e) {
+                logger.error("Could not stop http client.", e);
```

Comment:
Given we are closing its not that important, just null it out (please do that anyway) and create a new one next time.

Please do not log to error in this case, debug is enough.